src/asynk/async_worker.rs

```diff
@@ -106,4 +99,201 @@ where
             };
         }
     }
+    #[cfg(test)]
+    pub async fn run_tasks_until_none(&mut self) -> Result<(), Error> {
+        loop {
+            match self
+                .queue
+                .fetch_and_touch_task(&Some(self.task_type.clone()))
+                .await
+            {
+                Ok(Some(task)) => {
+                    self.sleep_params.maybe_reset_sleep_period();
+                    self.run(task).await?
+                }
+                Ok(None) => {
+                    return Ok(());
+                }
+                Err(error) => {
+                    error!("Failed to fetch a task {:?}", error);
+
+                    self.sleep().await;
+                }
+            };
+        }
+    }
+}
+
+#[cfg(test)]
+mod async_worker_tests {
```

Comment:
one more test can be added to test if tasks are being removed

Reply:
Done in [7bc47de](https://github.com/ayrat555/fang/pull/46/commits/7bc47dec117e3dc4d7957f48d5df0eb5185759ee)